packages/lesswrong/lib/vulcan-lib/callbacks.ts

```diff
@@ -370,22 +377,41 @@ function markCallbackStarted(description)
     pendingCallbackKey = 0;
   else
     pendingCallbackKey++;
-  pendingCallbacks[pendingCallbackKey] = true;
+  pendingCallbackKeys[pendingCallbackKey] = true;
+  
+  if (description in pendingCallbackDescriptions) {
+    pendingCallbackDescriptions[description]++;
+  } else {
+    pendingCallbackDescriptions[description] = 1;
+  }
+  
   return pendingCallbackKey;
 }
 
 // Record the fact that an async callback with the given ID has finished.
-function markCallbackFinished(id)
+function markCallbackFinished(id: number, description: string)
 {
   numCallbacksPending--;
-  delete pendingCallbacks[id];
+  delete pendingCallbackKeys[id];
+  
+  if (pendingCallbackDescriptions[description] <= 1) {
```

Comment:
I'm seeing this bug:
`2021-12-07T02:26:19.298Z: Callbacks in progress: lwevent.create.async(NaN), lwevents.new.async(NaN)`

I think it's because `undefined <= 1` so we are decrementing when the key doesn't exist, which results in the count being `NaN`.

Reply:
Good catch, I was indeed seeing `NaN`s in my tests and this is why.